### ISSUE TITLE
[Idea] Do stars or forks influence the statistics for Activity or Influence?

### ISSUE THREAD

**Issue by TommyLemon** [hypertrons/hypertrons-crx#530]
### Description

It doesn't seems to be that.
Well, I think the stars and forks should also be factors for the statistics of Influence.
And also, an account that relates to a famous/big company should influence more than other accounts.
Eg: If a Google employee starred a repo, the star he/she gave is more valuable than an account that relates to Baidu or even not relates to any company or organization. 

https://ossinsight.io/analyze/Tencent/APIJSON 

![FA2F41AC89A553D6B88F5E1D1D83E35C](https://user-images.githubusercontent.com/5738175/202859417-aa497459-f95e-4dd7-b5d4-3308ffa5aad1.jpg)

**Comment by tyn1998** [comment]
> It doesn't seems to be that because I noticed that some famous and popular repos have small numbers.

Yes, forks and stars at the moment have no influence on the caculation of `Activity` and `Influence`.

> Well, I think the stars and forks should also be factors for the statistics of hypertrons-crx's Influence due to they are also valuable. Yet they are much easier to get than PRs/issues. And you know many Chinese users haven't get used to use GitHub Issue, while they often chat(report bugs, give sugguestions, etc) in Wechat/QQ groups, and this results in smaller numbers of hypertrons-crx's Influence for many famous and popular Chinese repos when they are compared to American repos that have similar actual influence.

Good point of view! Hi @frank-zsy, how do you think about this?

> And also, an account that relates to a famous/big company/organization should influence more than other accounts.
Eg: If a Google employee starred a repo, the star he/she gave is more valuable than an account that relates to Baidu or even not relates to any company/organization.

That is a good advice!

Hi @TommyLemon, if you are interested in definitions of the metrics, you can visit [OpenDigger](https://github.com/X-lab2017/open-digger). You can get more information there :-)

**Comment by frank-zsy** [comment]
@TommyLemon Thanks for the discussion.

Metrics of open source repos are used for certain purpose. For `activity` and `influence`, I constrain the data to collaboration events, so stars and forks are not included since they are not really collaboration events.

Specially for `influence`, this [blog](https://blog.frankzhao.cn/how_to_measure_open_source_2) tells how to calculate the result by ***collaboration network*** which is used to measure the developers eco-system influence of a certain repo for a certain time period only with GitHub data.

Actually influence is not a good expression since it means `eigenvector centrality` in graph theory so it can be implemented in different ways with different data. In my own [research](https://github.com/X-lab2017/open-research/issues/28) I also use open source package dependency and developers' social relationships to construct the network. So we may change `influence` into `OpenRank` in the future to avoid misunderstanding.

> And also, an account that relates to a famous/big company/organization should influence more than other accounts.
Eg: If a Google employee starred a repo, the star he/she gave is more valuable than an account that relates to Baidu or even not relates to any company/organization.

It is true, and for OpenRank, actually the result is calculated by relationship data and a PageRank like algorithm, so the importance of each node(`repo` or `developer`) is different due to the position in the network, but still it is only in collaboration level.

**Comment by frank-zsy** [comment]
@TommyLemon And I am not really sure about how OSSInsight finds out the affiliation data of each developer, but there are really some legal risks to use the privacy data of developers like company affiliation or location.
